self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryInvoices.java

```diff
@@ -0,0 +1,277 @@
+            repoFullName,
+            username,
+            provider,
+            role
+        );
+        //get contract invoices.
+        List<Invoice> contractInvoices = invoices
+            .computeIfAbsent(contractId, k -> new ArrayList<>());
+        //get the active invoice id for the contract, if not create
+        //a new invoice.
+        Integer activeInvoiceId = activeIds.get(contractId);
+        if (activeInvoiceId == null) {
+            activeInvoiceId = ++this.idGenerator;
+            contractInvoices.add(new StoredInvoice(activeInvoiceId,
+                this.storage, contractId));
+            activeIds.put(contractId, activeInvoiceId);
+        }
+        //create the invoice task.
+        final int taskInvoiceId = activeInvoiceId;
+        final InvoiceTask invoiceTask = new InvoiceTask() {
+            @Override
+            public int invoiceId() {
+                return taskInvoiceId;
+            }
+            @Override
+            public Duration timeSpent() {
+                return timeSpent;
+            }
+            @Override
+            public Task task() {
+                return task;
+            }
+        };
+        //add the task to invoice.
+        invoiceTasks
+            .computeIfAbsent(activeInvoiceId, k -> new ArrayList<>())
+            .add(invoiceTask);
+
+        return invoiceTask;
+    }
+
+    @Override
+    public boolean isPaid(final int id) {
+        return payedIds
+            .values()
+            .stream()
+            .anyMatch(v -> v.contains(id));
+    }
+
+    @Override
+    public List<InvoiceTask> tasks(final int id) {
+        return invoiceTasks.getOrDefault(id, List.of());
+    }
+
+    @Override
+    public Iterator<Invoice> iterator() {
+        //invoices across all contracts
+        return invoices
+            .values()
+            .stream()
+            .flatMap(Collection::stream)
+            .iterator();
+    }
+
+    /**
+     * Simulates "payment" of an invoice.
+     * Used here to aid better representation of
+     * {@link InMemoryInvoices} state (as in "deactivating" the invoice).
+     *
+     * @param id Invoice id.
+     */
+    public void pay(final int id) {
+        final Map.Entry<Contract.Id, Integer> foundActiveEntry = activeIds
+            .entrySet()
+            .stream()
+            .filter(e -> e.getValue() == id)
+            .findFirst()
+            .orElse(null);
+        if (foundActiveEntry == null) {
+            throw new IllegalStateException("There is no active invoice "
+                + " with id " + id);
+        }
+        final Contract.Id contractId = foundActiveEntry.getKey();
+        final Integer invoiceId = foundActiveEntry.getValue();
+        payedIds.computeIfAbsent(contractId, k -> new ArrayList<>())
+            .add(invoiceId);
+        activeIds.remove(contractId);
+    }
+
+    /**
+     * Contract Invoices delegate. Manages invoices for a contract.
+     */
```

Comment:
@criske Can you just say "Invoices of a Contract", to follow the pattern of other similar classes?